src/components/OcButton.vue

```diff
@@ -1,13 +1,12 @@
 <template>
   <component
     :is="type"
-    :href="href"
-    :to="to"
+    :v-bind="properties"
     :type="submit"
     :aria-label="ariaLabel"
     :class="$_ocButton_buttonClass"
     :disabled="disabled"
-    @click="$_ocButton_onClick"
+    :v-on:click="$_ocButton_onClick"
```

Comment:
Misunderstanding. ;-) What I meant was that you introduce a `handlers` computed prop  and have `v-on="handlers"` here.
For the computed prop:
```js
const handlers = {}
if (this.type === "button") {
  handlers.click = this.$_ocButton_onClick
}
return handlers
```

This way we only have the click handler set at all when the component type is button. Otherwise is doesn't even have a click handler. You could even remove the type check within the click handler then.